gpt_index/embeddings/base.py

```diff
@@ -33,11 +33,27 @@ def __init__(self) -> None:
     def _get_query_embedding(self, query: str) -> List[float]:
         """Get query embedding."""
 
-    def get_query_embedding(self, query: str) -> List[float]:
+    def _combine_embeddings(self, embeddings: List[List[float]]) -> List[float]:
+        return list(np.array(embeddings).mean(axis=0))
+
+    def get_query_embedding(self, query: Union[str, List[str]]) -> List[float]:
         """Get query embedding."""
```

Comment:
tbh i'm not a huge fan of this polymorphism. Could this be a separate function e.g. agg_embedding_from_queries()?

Reply:
Do you mean overloading of the function name to handle both a `str` argument and `List[str]` argument? 

If so, yes I think I agree. Going to revise so they are separate methods. And in the embedding based query code, we can just call the `List[str]` version to support multi-document HyDE.